ts/utils/dates.ts

```diff
@@ -1,13 +1,41 @@
-import { format } from "date-fns";
+import { format as dateFnsFormat } from "date-fns";
+import en from "date-fns/locale/en";
+import it from "date-fns/locale/it";
+import I18n from "../i18n";
+
+type Locales = {
+  [index: string]: object;
+};
+
+const locales: Locales = { it, en };
+
+/**
+ * Extract language from locale coming from I18n library.
+ * @param locale string - has format: "language"_"country"
+ */
+const getLanguage = (locale: string) => locale.slice(0, 2);
 
 export function formatDateAsMonth(date: Date): string {
-  return format(date, "MMM");
+  return dateFnsFormat(date, "MMM", {
+    locale: locales[getLanguage(I18n.currentLocale())]
+  });
 }
 
 export function formatDateAsDay(date: Date): string {
-  return format(date, "DD");
+  return dateFnsFormat(date, "DD", { locale: locales[I18n.currentLocale()] });
 }
 
 export function formatDateAsReminder(date: Date): string {
-  return format(date, "YYYY-MM-DDTHH:mm:ss.SSS[Z]");
+  return dateFnsFormat(date, "YYYY-MM-DDTHH:mm:ss.SSS[Z]", {
```

Comment:
that format (ISO?) is [the default](https://date-fns.org/v1.9.0/docs/format#arguments) and it should be independent of the locales as all fields are numeric - so `formatDateAsReminder(date)` looks equivalent to `dateFnsFormat(date)`

Reply:
Right, so it would be better to remove changes about localization. Would you prefer me to do the same with "YYYY-MM-DDTHH:mm:ss.SSS[Z]" format as well?